datafusion/src/physical_plan/expressions/cast.rs

```diff
@@ -105,6 +105,7 @@ pub fn cast_with_error(
 ) -> Result<Box<dyn Array>> {
     let result = cast::cast(array, cast_type, options)?;
     if result.null_count() != array.null_count() {
+        println!("{result:?} : {array:?}");
```

Comment:
left over debug print?

Reply:
> Left one minor comment, the rest looks good to me. Thanks @Igosuki for this work. Sorry that I wasn't able to help with the migration last weekend because something urgent came up :(

Np, I'm actually fixing some of the issues in this PR